plugins/module_utils/_zoau_version_checker.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+# (c) Copyright IBM Corp. 2025
+# Apache License, Version 2.0 (see https://opensource.org/licenses/Apache-2.0)
+
+import traceback
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils.import_handler import ZOAUImportError
+
+try:
+    from zoautil_py import ZOAU_API_VERSION
+except Exception:
+    # Use ibm_zos_core's approach to handling zoautil_py imports so sanity tests pass
+    ZOAU_API_VERSION = ZOAUImportError(traceback.format_exc())
+
+
+CURRENT_MINIMUM_LEVEL = "1.3.0.0"
+
+IMPORT_ERROR_MESSAGE = f"Incompatible ZOAU api version found. Minimum supported version is v{CURRENT_MINIMUM_LEVEL}."
+
+
+def _check_zoau_version():
+    if isinstance(ZOAU_API_VERSION, str):
+        zoau_version = list(map(int, ZOAU_API_VERSION.split('.')))
+        min_version = list(map(int, CURRENT_MINIMUM_LEVEL.split('.')))
+        if not _zoau_version_greater_than_min(zoau_version, min_version):
+            raise ImportError(f"{IMPORT_ERROR_MESSAGE} Version found is {ZOAU_API_VERSION}")
+    else:
+        raise ImportError(IMPORT_ERROR_MESSAGE)
+
+
+def _zoau_version_greater_than_min(zoau_version, min_version):  # type: (list[int],list[int]) -> bool
+    for i in range(max(len(zoau_version), len(min_version))):
+        if zoau_version[i] > min_version[i]:
```

Comment:
Wouldn't this throw if one of them is shorter than the other eg 1.4.1 and 1.3.0.0?

Reply:
Yes it does. Im pretty sure the length is always 4 numbers (1.X.X.X) so I figured we'd be ok if I dont do the length check. I can change it to handle variable lengths if you want?

Reply:
Oh cool didnt know that existed, will change it to that

Reply:
Ok I changed it and it worked nicely until I got to the sanity tests. So I cant get past `ERROR: plugins/modules/aux_temp_storage.py:0:0: import-error: Exception attempting to import module for argument_spec introspection, 'No module named 'packaging''`

Im wondering if because its a separate pip package (https://packaging.pypa.io/en/stable/) its not always there by default. Im also pretty against adding it to the requirements.txt as I dont want to have to add more pip installable deps